graphics/src/AssimpLoader.cc

```diff
@@ -0,0 +1,722 @@
+      {
+        double alphaCutoff = mat->AlphaThreshold();
+        bool twoSided = mat->TwoSidedEnabled();
+        // Ignore return value, parameter unchanged if value is not set
+        assimpMat->Get(AI_MATKEY_GLTF_ALPHACUTOFF, alphaCutoff);
+        assimpMat->Get(AI_MATKEY_TWOSIDED, twoSided);
+        mat->SetAlphaFromTexture(true, alphaCutoff, twoSided);
+      }
+    }
+#endif
+  }
+#ifndef ASSIMP_COMPATIBILITY
+  // Edge case for GLTF, Metal and Rough texture are embedded in a
+  // MetallicRoughness texture with metalness in B and roughness in G
+  // Open, preprocess and split into metal and roughness map
+  ret = assimpMat->GetTexture(
+      AI_MATKEY_GLTF_PBRMETALLICROUGHNESS_METALLICROUGHNESS_TEXTURE,
+      &texturePath);
+  if (ret == AI_SUCCESS)
+  {
+    auto [texName, texData] = this->LoadTexture(_scene, texturePath,
+        this->GenerateTextureName(_scene, assimpMat, "MetallicRoughness"));
+    // Load it into a common::Image then split it
+    auto texImg =
+      texData != nullptr ? texData : std::make_shared<common::Image>(texName);
+    auto [metalTexture, roughTexture] =
+      this->SplitMetallicRoughnessMap(*texImg);
+    pbr.SetMetalnessMap(
+        this->GenerateTextureName(_scene, assimpMat, "Metalness"),
+        metalTexture);
+    pbr.SetRoughnessMap(
+        this->GenerateTextureName(_scene, assimpMat, "Roughness"),
+        roughTexture);
+  }
+  else
+  {
+    // Load the textures separately
+    ret = assimpMat->GetTexture(aiTextureType_METALNESS, 0, &texturePath);
+    if (ret == AI_SUCCESS)
+    {
+      auto [texName, texData] = this->LoadTexture(_scene, texturePath,
+          this->GenerateTextureName(_scene, assimpMat, "Metalness"));
+      pbr.SetMetalnessMap(texName, texData);
+    }
+    ret = assimpMat->GetTexture(
+        aiTextureType_DIFFUSE_ROUGHNESS, 0, &texturePath);
+    if (ret == AI_SUCCESS)
+    {
+      auto [texName, texData] = this->LoadTexture(_scene, texturePath,
+          this->GenerateTextureName(_scene, assimpMat, "Roughness"));
+      pbr.SetRoughnessMap(texName, texData);
+    }
+  }
+#endif
+  ret = assimpMat->GetTexture(aiTextureType_NORMALS, 0, &texturePath);
+  if (ret == AI_SUCCESS)
+  {
+    auto [texName, texData] = this->LoadTexture(_scene, texturePath,
+        this->GenerateTextureName(_scene, assimpMat, "Normal"));
+    // TODO(luca) different normal map spaces
+    pbr.SetNormalMap(texName, NormalMapSpace::TANGENT, texData);
+  }
+  // TODO(luca) Add support for emissive maps
+#ifndef ASSIMP_COMPATIBILITY
+  double value;
+  ret = assimpMat->Get(AI_MATKEY_METALLIC_FACTOR, value);
+  if (ret == AI_SUCCESS)
+  {
+    pbr.SetMetalness(value);
+  }
+  ret = assimpMat->Get(AI_MATKEY_ROUGHNESS_FACTOR, value);
+  if (ret == AI_SUCCESS)
+  {
+    pbr.SetRoughness(value);
+  }
+#endif
+  mat->SetPbrMaterial(pbr);
+  return mat;
+}
+
+//////////////////////////////////////////////////
+std::pair<std::string, ImagePtr> AssimpLoader::Implementation::LoadTexture(
```

Comment:
The test coverage report is in. The entire file is 77.6% covered, which is not bad, but we're aiming for >90%. This function isn't covered at all, can we test it?

Reply:
The coverage report shows up better to me when I'm logged into codecov:

https://app.codecov.io/gh/gazebosim/gz-common/pull/393

Reply:
I added a large test to get pretty much all the features tested other than the stray nullptr check that would require a fair bit of "creating invalid assets manually and make sure nothing goes badly".

Codecov still shows a large chunk of code not being tested, for example the whole `SplitMetallicRoughnessMap` function.
This function is private and is actually used and tested in a unit test that passes if the assimp version is above 5.0.0, specifically [here](https://github.com/gazebosim/gz-common/pull/393/files#diff-0fd2d29168662b4fdd425dcafa2a6d90a35ee01c90f533e4126fe509b4390a89R684-R687), however from the looks of the report it seems that codecov is running an older version of Assimp (maybe Ubuntu 20.04?) hence the whole chunk is `#ifndef`d away and not tested.
I could add an `#ifndef` to the function to remove dead code in Ubuntu 20.04 and increase the percentage of coverage but that would be pretty much only for the purpose of increasing coverage.

Reply:
Thanks, 88% is ok for now. We should look into either generating covarege from Jammy (moving `codecov_enabled` [here](https://github.com/gazebosim/gz-common/blob/main/.github/workflows/ci.yml)) or if it's something we're explicitly not testing, ignore it with `// LCOV_EXCL_START` and `// LCOV_EXCL_STOP`